File: 136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#

# @lc code=start
import functools
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        total = 0
        for num in nums:
            total ^= num
        return total
# @lc code=end

File: 2020/136.single-number.js
/*
 * @lc app=leetcode id=136 lang=javascript
 *
 * [136] Single Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
  return nums.filter(v=>nums.indexOf(v)==nums.lastIndexOf(v))[0]
};
// @lc code=end

